xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAEResamplePi.cpp

```diff
@@ -280,7 +280,7 @@ bool CActiveAEResamplePi::Init(uint64_t dst_chan_layout, int dst_channels, int d
 
   LOGTIME(3);
 
-  if (CSettings::Get().GetBool("videoplayer.usedisplayasclock") && CSettings::Get().GetInt("videoplayer.synctype") == SYNC_RESAMPLE)
+  if (CSettings::Get().GetBool("videoplayer.usedisplayasclock"))
```

Comment:
you take assumptions which may not hold in the future or even now. this broke because of a change in a part of the code which should not have a relationship to this. paplayer will make use of resampling too. 
check AESTREAM_FORCE_RESAMPLE. AE could pass this flag to resampler if required.